docs/design/014-did-doc/README.md

```diff
@@ -0,0 +1,590 @@
+![Alt](./create-key-2nd-device.svg "Create key on 2nd Edge Device")
+
+### Rotate a key by an Edge Device who is authenticated to do the update
+
+![Alt](./rotate-key-with-auth.svg "Rotate a key by an Edge Device who is authenticated to do the updatee")
+
+### Rotate a key by an Edge Device who is not authenticated to do the update
+
+![Alt](./rotate-key-without-auth.svg "Rotate a key by an Edge Device who is not authenticated to do the updatee")
+
+### A2A Communication
+
+![Alt](./a2a.svg "A2A Communication")
+
+## Identity structure
+
+On Indy SDK level DID-related identity has the following structure:
+
+- DID Method
+- Id string
+- Keys (private and public) on different edge devices
+  - Private keys are not shared between devices (the only possible exception is backup)
+- Authorization, authentication and discovery meta information:
+  - Public copy on ledger as DID DOC
+  - Optional local copy or cache
+
+## Design principles
+
+1. Each edge device stores the following information for every owned DID:
+    - Fully-qualified Did - Identifies data in storages
+    - Private and public keys associated with the DID from this edge device - in Wallet
+2. Each edge device can optionally store cached local copy of DID Doc for owned DID:
+    - The current DID Doc copy is used for signing/pack.
+    - The current DID Doc copy is synchronized between devices only through the Ledger (Public or microledger).
+    - DID Doc is never sent from one device to another.
+    - Local DID Doc copy is never updated explicitly.
+    - If a change to DID Doc is needed, the corresponding txn is created and sent to the Ledger.
+    - Different policies can be supported for updating the current DID DOC copy:
+      - The default one may just go to the Ledger every time before accessing the DID/DIDDoc. For example, if edge wants
+        to pack a message, it goes to the Ledger, gets the latest DID Doc, updates the local copy, and uses the local copy
+        for packing.
+      - Other policies may update it by timeout, or update by getting notifications from the Ledger/microledger
+3. In communication between Agents:
+    - DID Doc is never passed from one Agent (Alice) to another (Bob)
+    - DID is the only thing which is passed
+    - The receiver resolves the DID looking at the did method and going to the corresponding ledger (public or micro)
+    - This will allow communication between old agents (not supporting DID Doc) and new ones using the new approach
+4. The Ledger level is opaque. It doesn’t matter if this is public or micro ledger. Exact publishing and resolving strategy are
+   defined by a pluggable Resolvers and doesn’t affect workflows.
+5. To support DID Doc new transactions will be added to Indy Ledger:
+    - DID_DOC
+      - Consists of `did` and `doc`
+      - The ‘doc’ fields is a map of the form {field_id -> field} where field_id is a unique identifier of a field in the doc, and field is one of the fields (either array or a single-value)
```

Comment:
So no nesting?

Reply:
I think `field_id` can be a combined ID of multiple nested ids